apps/angular/src/core/pipes/no-empty.pipe.ts

```diff
@@ -0,0 +1,19 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+const NULL_VALUE = '-';
+
+/** A custom pipe to transform an empty string to a default value. */
+@Pipe({
+	name: 'noEmpty',
+	standalone: true,
+})
+export class NoEmptyPipe implements PipeTransform {
+	/**
+	 * Transform an empty value to a given default value.
+	 * @param value The given value.
+	 * @returns Return the given value or a default value if the value is empty.
+	 */
+	public transform(value: string | number | null): string | number {
+		return value != null && (value !== '' || typeof value === 'number' && !isNaN(value)) ? value : NULL_VALUE;
```

Comment:
Try not to chaining conditional operator when it's get too many cause it will be harder to read and understand, especially when condition get complicated. We can use early return for this.
```suggestion
		const isValueNan = typeof value === 'number' && isNaN(value);
		if (value == null || value === '' || isValueNan) {
			return NULL_VALUE;
		}

		return value;
```